portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -46,6 +45,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.getWriter().println(toJson(jsonObject));
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+
+    JSONObject jsonObject;
+    try {
+      jsonObject = new JSONObject();
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to create JSON object.", e);
+    }
+
+    response.setContentType("application/json;");
+    if (comment == null || comment.isEmpty()) {
+      jsonObject.put("error", "Bad request.");
+      response.getWriter().println(toJson(jsonObject));
```

Comment:
i think you need: `response.setStatus(SC_BAD_REQUEST)` too. HTTP error codes are an important part of building an API.